pipeline/daily.groovy

```diff
@@ -4,34 +4,43 @@ pipelineStages = load 'infrastructure/pipeline/daily/stages.groovy'
 
 def execute() {
   timestamps {
-    stage('Initialize') {
-      pipelineStages.initialize()
-    }
+    try {
+      stage('Initialize') {
+	pipelineStages.initialize()
+      }
 
-    node('builds_slave_label') {
-      lock(resource: "update-versions_workspace_$env.NODE_NAME") {
-        stage('Update packages versions') {
-          pipelineStages.updateVersions()
-        }
-
-        stage('Build packages') {
-          pipelineStages.buildPackages()
-        }
-
-        if (currentBuild.result != 'FAILURE') {
-          stage('Build ISO') {
-            pipelineStages.buildIso()
-          }
-        }
-
-        stage('Upload build artifacts') {
-          pipelineStages.uploadBuildArtifacts()
-        }
-
-        stage('Create symlinks') {
-          pipelineStages.createSymlinks()
-        }
+      node('builds_slave_label') {
+	lock(resource: "update-versions_workspace_$env.NODE_NAME") {
+	  stage('Update packages versions') {
+	    pipelineStages.updateVersions()
+	  }
+
+	  stage('Build packages') {
+	    pipelineStages.buildPackages()
+	  }
+
+	  if (currentBuild.result != 'FAILURE') {
+	    stage('Build ISO') {
+	      pipelineStages.buildIso()
+	    }
+	  }
+
+	  stage('Upload build artifacts') {
+	    pipelineStages.uploadBuildArtifacts()
+	  }
+
+	  stage('Create symlinks') {
+	    pipelineStages.createSymlinks()
+	  }
+	}
+      }
+    } catch (Exception exception) {
+      if (pipelineStages.shouldNotifyOnFailure()) {
+	stage('Notify failure') {
```

Comment:
I don't know if creating a stage for this is a good idea. If I recall correctly some tests I've made, this will make Jenkins workflow view lose information about stages that were not run.
For example, if it fails on 'Build packages', it will now show a column for 'Notify failure' and drop the columns for 'Build ISO', 'Upload build artifacts' and 'Create symlinks' from previous builds.